test/e2e/apps/job.go

```diff
@@ -361,6 +368,172 @@ var _ = SIGDescribe("Job", func() {
 			framework.ExpectEqual(pod.Status.Phase, v1.PodFailed)
 		}
 	})
+
+	ginkgo.It("should test the lifecycle of a job", func() {
+		one := int64(1)
+		jobName := "e2e-" + utilrand.String(5)
+		label := map[string]string{"e2e-job-label": jobName}
+		labelSelector := labels.SelectorFromSet(label).String()
+
+		ns := f.Namespace.Name
+		jobClient := f.ClientSet.BatchV1().Jobs(ns)
+
+		w := &cache.ListWatch{
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = labelSelector
+				return jobClient.Watch(context.TODO(), options)
+			},
+		}
+		jobsList, err := jobClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+		framework.ExpectNoError(err, "failed to list Pods")
+
+		ginkgo.By("Creating a suspended job")
+		job := e2ejob.NewTestJob("succeed", jobName, v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
+		job.Labels = label
+		job.Spec.Suspend = pointer.BoolPtr(true)
+		job, err = e2ejob.CreateJob(f.ClientSet, ns, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", ns)
+
+		ginkgo.By("Ensuring pods aren't created for job")
+		framework.ExpectEqual(wait.Poll(framework.Poll, wait.ForeverTestTimeout, func() (bool, error) {
+			pods, err := e2ejob.GetJobPods(f.ClientSet, ns, job.Name)
+			if err != nil {
+				return false, err
+			}
+			return len(pods.Items) > 0, nil
+		}), wait.ErrWaitTimeout)
```

Comment:
This kind of wait (wait for nothing to happen) is expensive and is already tested in another test. Do you really need to test it again here?

Reply:
Agree. Removed